✨feat: UI Driven MCP Mutations

dustinhealy opened the pull request:
Related to #7800 

## Summary

The updates include new API routes for MCP management, a restructuring of MCP-related types and forms, and a redesign of the MCP user interface originally prototyped for the AgentPanel. The functionality of the MCP Settings panel has been extended. Previously the panel was used to configure `customUserVars` in MCPs supplied through the `librechat.yaml`, and would only be visible in the sidenav when such an MCP existed in the YAML. The sidenav now includes a dedicated UI panel for MCP Servers that will continue to allow you to configure those customUserVars, as well as add new MCPs through the UI, which can be added to chats and agents. These new MCPs can be configured with custom headers and timeout specifications.Key changes are grouped below by theme.

- **Backend**: Implements new API routes for CRUD operations on MCP servers (`getMCPServers`, `getMCPServer`, `createMCPServer`, `updateMCPServer`, `deleteMCPServer`) in `api/server/routes/mcp.js`. 
- **Types**: Refactors MCP form/data types like `MCPMetadata` type with `customHeaders`, `requestTimeout`, `connectionTimeout`. Removes old borrowed Actions authentication fields and updates defaults to match new model.
- **Frontend UI**: Adds a new `MCPConfig` component for managing custom headers and timeouts when generating an MCP in the UI. Refactors the `MCPPanel` from Agents Panel to `MCPFormPanel` supporting full CRUD via hooks (`useCreateMCPMutation`, `useUpdateMCPMutation`, `useDeleteMCPMutation`).
- **Code Cleanup**: Updates imports and removes code related to the old Agent Builder MCP section.

## Change Type

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Testing

- UI manually tested locally.
- Backend unit tests for MCP endpoints added in `server.spec.ts`.

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [ ] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [ ] A pull request for updating the documentation has been submitted.

danny-avila commented on api/server/routes/mcp.js:
```diff
@@ -202,4 +209,44 @@ router.get('/oauth/status/:flowId', async (req, res) => {
+ * @route GET /api/mcp
+ * @returns {Array} Array of MCP servers
+ */
+router.get('/', requireJwtAuth, getMCPServers());
```
i think using factory functions here is unnecessary, please refactor to use it directly

dustinhealy committed:
✨ feat: Implement MCP Server Management API and UI Components

- Added API endpoints for managing MCP servers: create, read, update, and delete functionalities.
- Introduced new UI components for MCP server configuration, including MCPFormPanel and MCPConfig.
- Updated existing types and data provider to support MCP operations.
- Enhanced the side panel to include MCP server management options.
- Refactored related components and hooks for better integration with the new MCP features.
- Added tests for the new MCP server API functionalities.

danny-avila commented:
Closing, stale. New effort already in the works
